bugbug/tools/code_review.py

```diff
@@ -121,24 +134,109 @@ class HunkNotInPatchError(ModelResultError):
+       );
+     }
+     this.enabledStaticRules = rulesets;
++    this.#updateAllowAllRequestRules();
+   }
+
+   getSessionRules() {
+     return this.sessionRules.rules;
+   }
+
+   getDynamicRules() {
+     return this.dynamicRules.rules;
++  }
++
++  #updateAllowAllRequestRules() {
++    const filterAAR = rule => rule.action.type === "allowAllRequests";
++    this.hasRulesWithAllowAllRequests =
++      this.sessionRules.rules.some(filterAAR) ||
++      this.dynamicRules.rules.some(filterAAR) ||
++      this.enabledStaticRules.some(ruleset => ruleset.rules.some(filterAAR));
+   }
+ }
+
+ function getRuleManager(extension, createIfMissing = true) {
+   let ruleManager = gRuleManagers.find(rm => rm.extension === extension);""",
     },
     {
-        "file": "drs/control/Statistics.cpp",
-        "code_line": 25,
-        "comment": "The condition in the `if` statement is a bit complex and could be simplified for better readability. Consider extracting `!Components.isSuccessCode(status) && blockList.includes(ChromeUtils.getXPCOMErrorName(status))` into a separate function with a descriptive name, such as `isBlockedError`.",
+        "comment": {
+            "filename": "devtools/shared/network-observer/NetworkUtils.sys.mjs",
+            "start_line": 496,
+            "content": "The condition in the `if` statement is a bit complex and could be simplified for better readability. Consider extracting `!Components.isSuccessCode(status) && blockList.includes(ChromeUtils.getXPCOMErrorName(status))` into a separate function with a descriptive name, such as `isBlockedError`.",
+        },
+        "raw_hunk": """@@ -481,26 +481,21 @@
+     }
+   } catch (err) {
+     // "cancelledByExtension" doesn't have to be available.
+   }
+
+-  const ignoreList = [
+-    // This is emitted when the request is already in the cache.
+-    "NS_ERROR_PARSED_DATA_CACHED",
+-    // This is emitted when there is some issues around imgages e.g When the img.src
+-    // links to a non existent url. This is typically shown as a 404 request.
+-    "NS_IMAGELIB_ERROR_FAILURE",
+-    // This is emitted when there is a redirect. They are shown as 301 requests.
+-    "NS_BINDING_REDIRECTED",
++  const blockList = [
++    // When a host is not found (NS_ERROR_UNKNOWN_HOST)
++    "NS_ERROR_UNKNOWN_HOST",
+   ];
+
+   // If the request has not failed or is not blocked by a web extension, check for
+   // any errors not on the ignore list. e.g When a host is not found (NS_ERROR_UNKNOWN_HOST).
+   if (
+     blockedReason == 0 &&
+     !Components.isSuccessCode(status) &&
+-    !ignoreList.includes(ChromeUtils.getXPCOMErrorName(status))
++    blockList.includes(ChromeUtils.getXPCOMErrorName(status))
+   ) {
+     blockedReason = ChromeUtils.getXPCOMErrorName(status);
+   }
+
+   return { blockingExtension, blockedReason };""",
     },
 ]
 
+TEMPLATE_PATCH_FROM_HUNK = """diff --git a/{filename} b/{filename}
```

Comment:
Shouldn't the hunks be shown in the formatted patch format using the format_patch_set function? so that the examples are shown the same way the patches we ask the review for are

Reply:
The `format_patch_set()` function requires a parsed patch set, which we use ` PatchSet.from_string()` for. However, the hunk is a part of a patch (not a full patch) and does not show the file name. Thus, I used this template to show the hunk in a valid patch format.

Reply:
but then the TEMPLACE_PATCH_FROM_HUNK (without the diff line) could be used as the patch format then formatted with format_patch_set(), no?

Reply:
We do pass it to `format_patch_set()`. The following is the order:
```py
format_patch_set(
    PatchSet.from_string(
        TEMPLATE_PATCH_FROM_HUNK.format(
            filename,
            raw_hunk,
        )
    )
)
```

Reply:
you are right, missed the call on line 1052 and 1053. Looks good to me!